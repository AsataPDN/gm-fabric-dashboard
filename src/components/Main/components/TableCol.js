import styled from "styled-components";
import { PropTypes } from "prop-types";

import { errorColor } from "style/styleFunctions";

const TableCol = styled.div`
  ${props =>
    props.errorPercent ? `color: ${errorColor(props.errorPercent)}` : ""};
  flex: 1 1 12%;
  text-align: left;
`;

TableCol.propTypes = {
  errorPercent: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

export default TableCol;
