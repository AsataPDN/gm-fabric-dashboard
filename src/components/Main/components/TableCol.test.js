import React from "react";
import { shallow } from "enzyme";
import TableCol from "./TableCol";

describe("TableCol", () => {
  it("matches snapshot", () => {
    const aTableCol = shallow(<TableCol />);
    expect(aTableCol).toMatchSnapshot();
  });
});
