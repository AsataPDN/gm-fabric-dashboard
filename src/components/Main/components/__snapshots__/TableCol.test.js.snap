// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCol matches snapshot 1`] = `
.c0 {
  -webkit-flex: 1 1 12%;
  -ms-flex: 1 1 12%;
  flex: 1 1 12%;
  text-align: left;
}

<div
  className="c0"
/>
`;
